galasa-parent/dev.galasa.framework.api.resources/src/main/java/dev/galasa/framework/api/resources/routes/ResourcesRoute.java

```diff
@@ -95,54 +95,61 @@ public void processDataArray(JsonArray jsonArray, String action) throws Internal
-                    * will error if the property does not exist in CPS)
-                    * If the action is either "update" or "apply" and the property exists in CPS the updateProperty is set to true (update property)
-                    * If the action is equal to "apply" and the property does not exist in CPS the updateProperty is set to false (create property)
-                    * If the action is equal to "create" (force create) the updateProperty is set to false (create property, will error if the property exists in CPS)
-                    */
-                    boolean updateProperty = false;
-                    if ((updateActions.contains(action) && property.existsInStore()) || action.equals("update")){
-                        updateProperty = true;
+
+                    if (action.equals("delete")) {
+                        property.deletePropertyFromStore();
+                    }else{
+                        /*
+                        * The logic below is used to determine if the exclusive Not Or condition in property.setPropertyToStore 
+                        * (i.e. "the property exists" must equal to "is this an update action") will action the request or error
+                        *
+                        * Logic Table to Determine actions
+                        * If the action is equal to "update" (force update) the updateProperty is set to true (update property,
+                        * will error if the property does not exist in CPS)
+                        * If the action is either "update" or "apply" and the property exists in CPS the updateProperty is set to true (update property)
+                        * If the action is equal to "apply" and the property does not exist in CPS the updateProperty is set to false (create property)
+                        * If the action is equal to "create" (force create) the updateProperty is set to false (create property, will error if the property exists in CPS)
+                        */
+                        boolean updateProperty = false;
+                        if ((updateActions.contains(action) && property.existsInStore()) || action.equals("update")){
+                            updateProperty = true;
+                        }
+                        property.setPropertyToStore(galasaProperty, updateProperty);
                     }
-                    property.setPropertyToStore(galasaProperty, updateProperty);
+
                 }
-            }else{
+            } else {
                 ServletError error = new ServletError(GAL5027_UNSUPPORTED_API_VERSION, apiversion, expectedApiVersion);
                 throw new InternalServletException(error, HttpServletResponse.SC_BAD_REQUEST);
             }
-        }catch (InternalServletException i){
+        } catch (InternalServletException i){
             throw i;
-        }catch (NullPointerException n){
+        } catch (NullPointerException n){
```

Comment:
We need a way of avoiding catching these surely. Catching NPEs is a recipe for disaster.